libs/partners/anthropic/langchain_anthropic/chat_models.py

```diff
@@ -104,18 +104,15 @@ def _merge_messages(
     for curr in messages:
         curr = curr.copy(deep=True)
         if isinstance(curr, ToolMessage):
-            if isinstance(curr.content, str):
-                curr = HumanMessage(  # type: ignore[misc]
-                    [
-                        {
-                            "type": "tool_result",
-                            "content": curr.content,
-                            "tool_use_id": curr.tool_call_id,
-                        }
-                    ]
-                )
-            else:
-                curr = HumanMessage(curr.content)  # type: ignore[misc]
+            curr = HumanMessage(  # type: ignore[misc]
+                [
+                    {
+                        "type": "tool_result",
+                        "content": curr.content,
```

Comment:
we should check that curr.content isn't already a list with tool_result in it

Reply:
good call. made this change and added a test case.